contracts/SnappAuction.sol

```diff
@@ -187,6 +195,51 @@ contract SnappAuction is SnappBase {
         }
     }
 
+    function auctionSolutionBid(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 proposedStateRoot,
+        uint proposedObjectiveValue
+        // bytes32 _orderHash,
+        // bytes32 _standingOrderIndex          // should ensure acting on correct order set
+    ) public {
+        require(
+            slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0,
+            "Previous auction not yet resolved!"
+        );
+        // Verify state root and order hash before continuing.
+        require(coreData.stateRoots[stateIndex()] == _currStateRoot, "Incorrect state root");
+        // TODO - verify acting on correct order hash.
+        // require(
+        //     calculateOrderHash(slot, _standingOrderIndex) == _orderHash,
+        //     "Order hash doesn't agree"
+        // );
+
+        // Ensure that auction batch is inactive, unprocessed and in correct phase for bidding
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");  // This is a bit redundant.
```

Comment:
How is this redundant? If the auction is between 3-6 minutes from starting it cannot be applied yet? Let's remove this check then.

Reply:
If I remember correctly, this is redundant because the auction could not have already been applied if it is in the correct phase for bidding. I had left it to make sure I hadn't missed any cases. We will find that this check is absolutely necessary during the second phase!

Reply:
Turns out it was not redundant and this was caught by our unit tests!